gulpfile.mjs

```diff
@@ -2344,6 +2312,18 @@ gulp.task(
   })
 );
 
+gulp.task(
+  "typestest",
+  gulp.series(setTestEnv, "dist", function runTypesTest(done) {
```

Comment:
Do we *need* to run `gulp dist` as part of this, since that basically builds *everything* which will slow down the tests *a lot* just for the typescript definitions?

Reply:
If we do this I think it needs to be `dist-pre` at least because `dist` is basically `dist-pre` plus committing which fails in the CI currently. I would also like to see the shortest runtime here, but I can also see that testing against the real distribution is perhaps better, so I'm not sure what's best here. If the problem from the issue can be reproduced without `dist-pre` (so basically with the current code) that would be preferable for me too.

Reply:
If we absolutely must use e.g. `dist-pre` here, then I think that this really should be moved into *its own* workflow separate from the regular GitHub Actions tests and only running in e.g. Node 18.